meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -94,12 +97,21 @@ public void create(String key, Object data, MetaClientInterface.EntryMode mode)
 
   @Override
   public void createWithTTL(String key, T data, long ttl) {
-    throw new UnsupportedOperationException("TTL nodes aren't yet supported.");
+    try{
+      _zkClient.createPersistentWithTTL(key, data, ttl);
+    } catch (ZkException e) {
+      throw translateZkExceptionToMetaclientException(e);
+    }
   }
 
   @Override
   public void renewTTLNode(String key) {
-    throw new UnsupportedOperationException("TTL nodes aren't yet supported.");
+    if (!_zkClient.exists(key)) {
+      throw new MetaClientNoNodeException("Node at path " + key + " does not exist.");
+    }
+
+    T oldData = _zkClient.readData(key, _zkClient.getStat(key));
```

Comment:
I think read here would validate if node exists. No need to check exists

Reply:
You're right, will fix. Thanks!